test/Test/Pos/Ssc/GodTossing/Toss/BaseSpec.hs

```diff
@@ -0,0 +1,573 @@
+    description_checksGoodCommsPayload =
+        "successfully checks payload of commitments with a proper epoch index"
+    description_checksBadCommsPayload =
+        "unsuccessfully checks payload of commitments with the right exception for each\
+        \ failure case"
+    description_checksGoodOpensPayload =
+        "successfully checks payload of openings when the global state is correct"
+    description_checksBadOpeningsPayload =
+        "unsuccessfully checks payload of openings with the right exception for each\
+        \ failure case"
+    description_checksGoodSharesPayload =
+        "successfully checks payload of shares"
+    description_checksBadSharesPayload =
+        "unsuccessfully checks payload of shares with the right exception for each\
+        \ failure case"
+    description_checksGoodCertsPayload =
+        "successfully checks valid payload of VSS certificates"
+    description_checksBadCertsPayload =
+        "unsuccessfully checks payload of certificates with the right exception for each\
+        \ failure case"
+
+verifiesOkComm :: CommitmentOpening -> Bool
+verifiesOkComm CommitmentOpening{..} =
+    verifyCommitment coCommitment
+
+notVerifiesBadComm :: BadCommitment -> Bool
+notVerifiesBadComm (getBadComm -> badComm) =
+    not . verifyCommitment $ badComm
+
+verifiesOkCommSig :: SecretKey -> Commitment -> EpochIndex -> Bool
+verifiesOkCommSig sk comm epoch =
+    let commSig = (toPublic sk, comm, sign SignCommitment sk (epoch, comm))
+    in verifyCommitmentSignature epoch commSig
+
+notVerifiesBadCommSig :: BadSignedCommitment -> EpochIndex -> Bool
+notVerifiesBadCommSig (getBadSignedC -> badSignedComm) epoch =
+    not $ verifyCommitmentSignature epoch badSignedComm
+
+verifiesOkOpening :: CommitmentOpening -> Bool
+verifiesOkOpening CommitmentOpening{..} =
+    verifyOpening coCommitment coOpening
+
+notVerifiesBadOpening :: BadCommAndOpening -> Bool
+notVerifiesBadOpening (getBadCAndO -> badCommsAndOp) =
+    not . uncurry verifyOpening $ badCommsAndOp
+
+emptyPayload
+    :: Monoid container
+    => (container -> ExceptT e PureToss a)
+    -> MultiRichmenStake
+    -> GtGlobalState
+    -> Bool
+emptyPayload pureToss mrs gtgs =
+    isRight $ tossRunner mrs gtgs $ pureToss mempty
+
+newtype GoodCommsPayload = GoodComms
+    { getGoodComms :: (EpochIndex, GtGlobalState, CommitmentsMap, MultiRichmenStake)
+    } deriving (Show, Eq)
+
+instance Arbitrary GoodCommsPayload where
+    arbitrary = GoodComms <$> do
+        -- These fields won't be needed for anything, so they can be entirely arbitrary.
+        _gsOpenings <- arbitrary
+        _gsShares <- arbitrary
+
+        -- The epoch used in the tests is generated separately to make sure it exists.
+        (epoch, m) <- arbitrary :: Gen (EpochIndex, MultiRichmenStake)
+        richmen <- getValid <$> (arbitrary :: Gen (ValidRichmenStake GenesisMpcThd))
+        let richmenIds = HM.keys richmen
+            mrs = HM.insert epoch richmen m
```

Comment:
I think it doesn't make sense to generate `mrs` with more than one epoch here. It may slow tests down a little bit.

If you really want to test that adding richmen for other epochs doesn't spoil anything, you can create `mrs` for two epochs, for example.